app/src/main/java/ch/epfl/sdp/drone3d/ui/map/MissionInProgressActivity.kt

```diff
@@ -298,6 +299,32 @@ class MissionInProgressActivity : BaseMapActivity() {
         }
     }
 
+    private fun createDroneKeeperObserver(droneData: DroneData) {
+        // Create an observer assuring that the drone stays within [MAX_DIST_TO_USER] meters of the user
+        // and stay visible for the user
+        createObserver(droneData.getPosition()) {
+            it?.let {
+                if (locationService.isLocationEnabled() && locationService.getCurrentLocation() != null) {
+                    val distanceUser: Double = it.distanceTo(locationService.getCurrentLocation()!!)
+                    val maxDistance: Double = if (MAX_DIST_TO_USER > weatherReport.value!!.visibility) {
+                        weatherReport.value!!.visibility.toDouble()
+                    } else {
+                        MAX_DIST_TO_USER
+                    }
+
+                    if (distanceUser > maxDistance && !droneData.isMissionPaused().value!!) {
```

Comment:
Is it really safe to use value!! here without any check for null values?

Reply:
When droneData is set when a simulation is connected, all the MutableLiveData are set to a default value, so it shouldn't be a problem